pkg/models/organization_model.go

```diff
@@ -35,20 +36,46 @@ func (om *OrganizationModel) FindByID(ctx context.Context, id primitive.ObjectID
 }
 
 func (om *OrganizationModel) InsertOne(ctx context.Context, record *OrganizationRecord) (primitive.ObjectID, error) {
+	record.ID = primitive.NewObjectID()
 	result, err := om.collection.InsertOne(ctx, record)
 	if err != nil {
-		return primitive.ObjectID{}, err
+		return primitive.NilObjectID, err
 	}
 
 	objectID, ok := result.InsertedID.(primitive.ObjectID)
 
 	if !ok {
-		return primitive.ObjectID{}, errors.New("unable to assert type of objectID")
+		return primitive.NilObjectID, errors.New("unable to assert type of objectID")
 	}
 
 	return objectID, nil
 }
 
+func (om *OrganizationModel) UserHasReadPermission(
```

Comment:
I don`t think this should be here right @LuisOw

Reply:
Then move it to ``api_utils.go`` or create a new helper file.

Reply:
Actually, add the ```if !isMember {
		return apiutils.ErrReadPermissionDenied
	}
```
to the similar function on ``feature_flag_modelgo`` and move that to another file